### PR TITLE
Back end endpoints for MetaData

### PR DESCRIPTION
**Pull request by manuel-delvillar**
# Security
- Add  `@EnableGlobalMethodSecurity` to use `@PreAuthorize`
- Use  `@PreAuthorize` for security in controllers

# Job Codes
## Available for all authenticated users
- Get all job codes with pagination
- Get job code by ID

## Available for admin role
- Create one or more job codes
- Update a job code
- Delete a job code

# Named References
## Available for all authenticated users
- Get all named references with pagination
- Get named reference by ID

## Available for admin role
- Create one or more named references
- Update a named reference
- Delete a named reference

<br>

> Some request contains a little logic to save and return data, but important changes are related to match with the "open api file" in PR #391

### CODE REVIEW

**Review by jchavez137 — CHANGES_REQUESTED** [review]
Looks good to me, remove only the star imports and commented code in SecurityConfig

**Comment by JohnKallies** [comment]
Is there any test coverage in this change?

**Comment by manuel-delvillar** [comment]
@JohnKallies I am going to add some test for controller.

**Review by jchavez137 — APPROVED** [review]

File: api/src/main/kotlin/edu/wgu/osmt/api/model/JobCodeUpdate.kt [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    @JsonProperty("targetNodeName")
+    val targetNodeName: String? = null,
+    @JsonProperty("frameworkName")
+    val framework: String? = null,
+    @JsonProperty("level")
+    val level: JobCodeLevel? = null,
+    @JsonProperty("parents")
+    val parents: List<JobCodeUpdate>? = null
+    // @JsonProperty("description")
+    // val description: String = ""
```
Comment (Corpratespaz) [inline api/src/main/kotlin/edu/wgu/osmt/api/model/JobCodeUpdate.kt:20]: Why is this commented out? Is it not required, or should it be included?

File: api/src/main/kotlin/edu/wgu/osmt/jobcode/JobCodeRepository.kt [hunk 2]
```diff
@@ -52,6 +54,20 @@ class JobCodeRepositoryImpl: JobCodeRepository {
...
     }
 
+    override fun createFromApi(jobCodes: List<JobCodeUpdate>): List<JobCodeDao> {
+        return jobCodes.map { jobCodeUpdate ->
+            dao.new {
+                this.code = jobCodeUpdate.code
+                this.framework = jobCodeUpdate.framework
+                this.name = jobCodeUpdate.targetNodeName
+                this.creationDate = LocalDateTime.now(ZoneOffset.UTC)
+                // this.description = jobCodeUpdate.description
```
Comment (Corpratespaz) [inline api/src/main/kotlin/edu/wgu/osmt/jobcode/JobCodeRepository.kt:64]: Is this line being saved for a follow-up admin screen ticket, or should it be removed?

Comment (manuel-delvillar) [reply api/src/main/kotlin/edu/wgu/osmt/jobcode/JobCodeRepository.kt]: I removed the line since also was removed from `JobCodeUpdate`

**Pull request merged**